src/Language/PureScript/Kinds.hs

```diff
@@ -18,8 +20,8 @@ data Kind
   | Row Kind
   -- | Function kinds
   | FunKind Kind Kind
-  -- | Type-level strings
-  | Symbol
+  -- | A named kind
+  | NamedKind (Qualified (ProperName 'KindName))
```

Comment:
I wonder if `Star` and `Bang` should be moved into `Prim` too? Eventually, I would like `Bang` to be moved into `purescript-eff`, but in order to do that, we need to introduce a name for it, other than `!` (perhaps `Effect`?). So how about we move it into `Prim` now, allow both `!` and `Effect` in the parser, and then move it out to `purescript-eff` in 0.11?